jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -841,6 +841,138 @@ YUI.add('juju-controller-api', function(Y) {
+      });
+      this._send_rpc({
+        type: 'Cloud',
+        request: 'UserCredentials',
+        params: {'user-clouds': userClouds}
+      }, handler);
+    },
+
+    /**
+      Return the specified cloud credentials for each tag, minus secrets.
+
+      @method getCloudCredentials
+      @param {Array} tags The Juju tags of the credentials, each one being a
+        string, for instance "cloudcred-google_dalek@local_google". Tags for
+        credentials are usually retrieved by calling the
+        getTagsForCloudCredentials method (see above).
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive two parameters: an error (as a
+        string describing the problem) and an object mapping credentials tags
+        to corresponding cloud credentials info. If no errors occur, the error
+        parameter is null. Otherwise, in case of errors, the second argument is
+        an empty object. Credentials info is returned as objects with the
+        following fields:
+        - err: a possible credentials specific error, in which case all
+          subsequent fields are omitted;
+        - name: the cloud credentials name, like "google_dalek@local_google";
+        - authType: the authentication type (as a string, like 'jsonfile');
+        - attrs: non-secret credential values as an object mapping strings to
+          strings. Keys there are based on the cloud type;
+        - redacted: a list of names of redacted attributes.
+        If no errors occur, error parameters are null. Otherwise, in case of
+        errors, the second argument is an empty object.
+    */
+    getCloudCredentials: function(tags, callback) {
+      // Decorate the user supplied callback.
+      const handler = function(userCallback, data) {
+        if (!userCallback) {
+          console.log('data returned by Cloud.Credential API call:', data);
+          return;
+        }
+        if (data.error) {
+          userCallback(data.error, {});
+          return;
+        }
+        const results = data.response.results;
+        if (!results) {
+          userCallback(null, {});
+          return;
+        }
+        const credentials = results.reduce((prev, result, index) => {
+          const tag = tags[index];
+          const err = result.error && result.error.message;
+          if (err) {
+            prev[tag] = {err: err};
+            return prev;
+          }
+          const entry = result.result;
+          prev[tag] = {
+            name: tag.slice('cloudcred-'.length),
+            authType: entry['auth-type'] || '',
+            attrs: entry.attrs || {},
+            redacted: entry.redacted || []
+          };
+          return prev;
+        }, {});
+        userCallback(null, credentials);
+      }.bind(this, callback);
+      // Send the API request.
+      if (!tags.length) {
+        tags = [];
+      }
+      const entities = tags.map(function(tag) {
```

Comment:
As I mentioned before about implicit returns I think it's cases like this where it really shines.
```JavaScript
const entities = tags.map(tag => ({tag}));
```
In this case I used two new tricks. First because you're returning an object we have to wrap the curlies in a parens else it'll think it's the opening brackets to the function. The second is that if the key and value are the same string then you can shorten it to just `{tag}` and it'll expand out internally to `{tag: tag}`.